airflow/utils/context.py

```diff
@@ -274,14 +279,27 @@ def __getitem__(self, key: int | str | Asset | AssetAlias) -> LazyAssetEventSele
             obj = self._inlets[key]
             if not isinstance(obj, (Asset, AssetAlias)):
                 raise IndexError(key)
+
+            if isinstance(obj, AssetRef):
+                obj = _fetch_active_assets_by_name([obj.name])
```

Comment:
I wonder if this is a performance issue… when is `__getitem__` called?

Reply:
Yes, it might. I was thinking of whether we should fetch all the AssetRef when initializing. But that's kinda contradict the idea of lazy loading. Or maybe we could just load all the AssetRef when the first is loaded?

Reply:
Yeah maybe it’s easier to just load all AssetRef all at once.

Reply:
just updated to load it into `_assets`